compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/desugar/Desugar.java

```diff
@@ -4134,6 +4147,7 @@ public void visit(BLangArrowFunction bLangArrowFunction) {
 
         lambdaFunction.function.pos = bLangArrowFunction.pos;
         lambdaFunction.function.body.pos = bLangArrowFunction.pos;
+        lambdaFunction.capturedClosureEnv = env;
```

Comment:
Shall we add a comment here explaining why we're simply assigning `env` instead of a clone?

Reply:
The reasoning is by the time desugar phase runs, the program is guaranteed to be semantically correct. Which means in the lambda function, there are no references to out of scope symbols. Therefore, we can simply consider the enclosing env as the captured closure env as well.

Reply:
Added a comment